src/wire/Socket.ts

```diff
@@ -249,10 +249,12 @@ export class InteractiveSocket extends EventEmitter {
             return;
         }
 
-        this.state = SocketState.Closing;
-        this.socket.close(1000, 'Closed normally.');
-        this.queue.forEach(packet => packet.cancel());
-        this.queue.clear();
+        if (this.state !== SocketState.Idle) {
+            this.state = SocketState.Closing;
+            this.socket.close(1000, 'Closed normally.');
+            this.queue.forEach(packet => packet.cancel());
+            this.queue.clear();
+        }
```

Comment:
Probably want to set the state to `Closed` if the conditional fails, for cleanliness. Not sure if anything actually looks at it.